presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -246,12 +247,14 @@ private PlanNodeStatsEstimate filterByAuxiliaryClause(PlanNodeStatsEstimate stat
                 .setNullsFraction(0)
                 .setStatisticsRange(intersect)
                 .setDistinctValuesCount(retainedNdv)
+                .setHistogram(addConjunction(leftStats.getHistogram(), intersect))
```

Comment:
Histogram should always be non-null. If it was supposed to be nullable I would add an `@Nullable` annotation.

If a histogram is not set in `VariableStatisticsEstimate.Builder`, then it will default to a `UniformDistributionHistogram` with bounds equivalent to the low/high value set in the builder. By default `lowValue` is -inf, `maxValue` is +inf if stats are not available. Otherwise, if they are it will use the correct low/high bounds in the uniform histogram. I can add a `requireNonNull` check in the constructor to ensure it is never null.